tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -838,26 +829,33 @@ function Get-AzureRmProfile
     if ($ListAvailable.IsPresent)
     {
       Write-Verbose "Getting all the profiles available for install"
-      Get-ProfilesAvailable $ProfileMap
+      foreach ($profile in ($ProfileMap | get-member -MemberType NoteProperty).Name)
+      {
+        $profileObj = $ProfileMap.$profile
+        $profileObj | Add-Member -MemberType NoteProperty -Name "ProfileName" -Value $profile
```

Comment:
Are we pushing $profileObj to the output stream multiple times?  Is this intentional?